efficientdet/keras/train_lib.py

```diff
@@ -843,3 +849,61 @@ def test_step(self, data):
     loss_vals['reg_l2_loss'] = reg_l2_loss
     loss_vals['loss'] = total_loss + tf.cast(reg_l2_loss, loss_dtype)
     return loss_vals
+
+
+class EfficientDetNetTrainHub(EfficientDetNetTrain):
+  """EfficientDetNetTrain for Hub module."""
+
+  def __init__(self, config, hub_module_url, name=''):
+    super(efficientdet_keras.EfficientDetNet, self).__init__(name=name)
+    self.config = config
+    self.hub_module_url = hub_module_url
+    self.base_model = hub.KerasLayer(hub_module_url, trainable=True)
+
+    # class/box output prediction network.
+    num_anchors = len(config.aspect_ratios) * config.num_scales
+
+    if config.separable_conv:
+      conv2d_layer = functools.partial(
+          tf.keras.layers.SeparableConv2D, depth_multiplier=1)
+    else:
+      conv2d_layer = tf.keras.layers.Conv2D
+    self.classes = conv2d_layer(
+        config.num_classes * num_anchors,
```

Comment:
Initializers are different from original one.

Reply:
It seems to be the same as before: https://github.com/google/automl/blob/890eaddc0963701e515de06a11a662d1f1027770/efficientdet/keras/efficientdet_keras.py#L426

Notably, the last convnet for class prediction is special in order to enable focal loss.

Reply:
```pointwise_initializer``` and ```depthwise_initializer``` are ```tf.initializers.variance_scaling()``` in ClassNet, should it be changed? It's better to use ClassNet and BoxNet whose are defined in efficientnet_keras.py

Reply:
Good point. I am going to merge this for now, and update it later.